test/envtest/reconciler_setupwithmanager_test.go

```diff
@@ -71,10 +93,24 @@ var konnectGatewayControlPlaneTestCases = []konnectEntityReconcilerTestCase{
 							Name: "auth",
 						},
 					},
+					CreateControlPlaneRequest: sdkkonnectcomp.CreateControlPlaneRequest{
+						Name:        "cp-1",
+						Description: lo.ToPtr("test control plane 1"),
+					},
 				},
 			}
 			require.NoError(t, cl.Create(ctx, cp))
 		},
+		mockExpectations: func(sdk *ops.MockSDKWrapper, ns *corev1.Namespace) {
+			sdk.ControlPlaneSDK.EXPECT().CreateControlPlane(mock.Anything, sdkkonnectcomp.CreateControlPlaneRequest{
+				Name:        "cp-1",
+				Description: lo.ToPtr("test control plane 1"),
+			}).Return(&sdkkonnectops.CreateControlPlaneResponse{
+				ControlPlane: &sdkkonnectcomp.ControlPlane{
+					ID: "12345",
+				},
+			}, nil)
+		},
```

Comment:
I've just realized that for those expectations to be effective you need to `defer`/`t.Cleanup()` `.AssertExpectations(t)` or even better use the mock constructors which have these called via cleanup:

https://github.com/Kong/gateway-operator/blob/0ace883118c7f61f8a0ccae1518721d0845ef9f6/controller/konnect/ops/controlplane_mock.go#L253-L260

Reply:
Added `mock.AssertExpectations` in test cases.